### PR TITLE
Expose "raw" property type on Expr class

### PR DESCRIPTION
**Pull request by aleclarson**

### CODE REVIEW

File: src/types/Expr.d.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export default class Expr {
   constructor(obj: object)
+  readonly raw: object
```
Comment (marrony) [inline src/types/Expr.d.ts]: That property is supposed to be private, actually you shouldn’t handling `Expr` objects, this is part of fauna DSL and is meant to be used with `client.query(...)` to be evaluated on the server side

Comment (aleclarson) [reply src/types/Expr.d.ts]: I'm using it to check if an `Expr` is of a specific type, like this:
```ts
  q.LoadCollection = (e: string | Expr, opts?: object) => {
    if (typeof e === 'string') {
      e = q.Collection(e)
    }
    return q.Paginate('collection' in e.raw ? q.Documents(e) : e, opts)
  }

q.LoadCollection('foo')
```
I don't see the harm in doing so?

Comment (marrony) [reply src/types/Expr.d.ts]: if you change this function to:
```javascript
  q.LoadCollection = (e: string | Expr, opts?: object) => {
    if (typeof e === 'string') {
      e = q.Documents(q.Collection(e))
    }
    return q.Paginate(e, opts)
  }

q.LoadCollection('foo')
```
you won't need to inspect `raw` property on `Expr`

**Comment by aleclarson** [comment]
Just realized that duck-typing the `raw` object isn't foolproof, since `q.Var` queries are opaque in what they resolve to.

**Pull request closed without merging**